java/src/main/java/com/adhoc/flight/QueryRunner.java

```diff
@@ -119,6 +119,10 @@ static class CommandLineArguments {
         description = "The specific engine to run against.")
     public String engine;
 
+    @Parameter(names = {"-projectId", "--projectId"},
```

Comment:
Maybe we can specify here that the `projectId` is optional.

Reply:
By default most of the parameters are optional. I suppose it's better than to mark the required only.

Reply:
But most of the parameters are optional. So I will leave it unchanged and by default as optional.